pkg/file/splitter/internal/job.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package internal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash"
+	"os"
+
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bmt"
+	bmtlegacy "github.com/ethersphere/bmt/legacy"
+	"golang.org/x/crypto/sha3"
+)
+
+// hashFunc is a hasher factory used by the bmt hasher
+func hashFunc() hash.Hash {
+	return sha3.NewLegacyKeccak256()
+}
+
+// SimpleSplitterJob encapsulated a single splitter operation, accepting blockwise
+// writes of data whose length is defined in advance.
+//
+// After the job is constructed, Write must be called with up to ChunkSize byte slices
+// until the full data length has been written. The Sum should be called which will
+// return the SwarmHash of the data.
+//
+// Called Sum before the last Write, or Write after Sum has been called, may result in
+// error and will may result in undefined result.
+type SimpleSplitterJob struct {
+	ctx        context.Context
+	store      storage.Storer
+	spanLength int64    // target length of data
+	length     int64    // number of bytes written to the data level of the hasher
```

Comment:
i think these two should be `uint64`

Reply:
It seems to me that anything `length` in `golang` is int.